File: src/minecraft/monnef/jaffas/technic/item/ItemSwordTechnic.java
/*
 * Jaffas and more!
 * author: monnef
 */

package monnef.jaffas.technic.item;

import monnef.core.MonnefCorePlugin;
import monnef.jaffas.food.item.ItemJaffaSword;
import monnef.jaffas.technic.JaffasTechnic;
import monnef.jaffas.technic.Reference;
import net.minecraft.block.Block;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.EnumToolMaterial;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;

import java.util.List;

public class ItemSwordTechnic extends ItemJaffaSword {
    public ItemSwordTechnic(int id, int textureOffset, EnumToolMaterial material) {
        super(id, textureOffset, material);
        setCreativeTab(JaffasTechnic.instance.creativeTab);
        setSecondCreativeTab(CreativeTabs.tabCombat);
        setSheetNumber(3);
    }

    @Override
    public String getModName() {
        return Reference.ModName;
    }

    @Override
    public void getSubItems(int par1, CreativeTabs par2CreativeTabs, List par3List) {
        par3List.add(new ItemStack(par1, 1, 0));
        if (MonnefCorePlugin.debugEnv) par3List.add(new ItemStack(par1, 1, getMaxDamage() - 10));
    }

    @Override
    public boolean onItemUse(ItemStack stack, EntityPlayer player, World world, int x, int y, int z, int side, float par8, float par9, float par10) {
        if (!nearlyDestroyed(stack)) {
            int blockId = world.getBlockId(x, y, z);
            if (blockId == Block.web.blockID) {
                if (!world.isRemote) {
                    damageTool(2, player, stack);
                    world.destroyBlock(x, y, z, true);
                }
                return true;
            } else {
                return false;
            }
        } else {
            return false;
        }
    }
}

File: src/main/scala/monnef/technic/item/ItemSwordTechnic.scala
package monnef.jaffas.technic.item

import monnef.jaffas.food.item.{ItemJaffaTool, ItemJaffaSword}
import monnef.jaffas.technic.JaffasTechnic
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.{EnumToolMaterial, ItemStack}
import monnef.core.MonnefCorePlugin
import net.minecraft.block.Block
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World
import monnef.jaffas.technic.common.IconDescriptorTechnic

class ItemSwordTechnic(_id: Int, _tex: Int, _mat: EnumToolMaterial) extends ItemJaffaSword(_id, _tex, _mat) with IconDescriptorTechnic {
  setCreativeTab(JaffasTechnic.instance.creativeTab)
  setSecondCreativeTab(CreativeTabs.tabCombat)

  override def getSubItems(par1: Int, par2CreativeTabs: CreativeTabs, par3List: java.util.List[_]) {
    val l = par3List.asInstanceOf[java.util.List[ItemStack]]
    l.add(new ItemStack(par1, 1, 0))
    if (MonnefCorePlugin.debugEnv) l.add(new ItemStack(par1, 1, getMaxDamage - 10))
  }

  override def onItemUse(stack: ItemStack, player: EntityPlayer, world: World, x: Int, y: Int, z: Int, side: Int, par8: Float, par9: Float, par10: Float): Boolean = {
    if (!ItemJaffaTool.nearlyDestroyed(stack)) {
      val blockId: Int = world.getBlockId(x, y, z)
      if (blockId == Block.web.blockID) {
        if (!world.isRemote) {
          damageTool(2, player, stack)
          world.destroyBlock(x, y, z, true)
        }
        true
      }
      else false
    }
    else false
  }
}

